### PR TITLE
Add posix_spawn

### PR DESCRIPTION
**Pull request by boricj**
This adds the posix_spawn family of functions (implementation based off newlib's). It should also fix configure scripts which were fooled by the presence of spawn.h despite the lack of posix_spawn support.

### CODE REVIEW

**Comment by dcvmoole** [comment]
At a cursory glance the patch looks good to me, and I really appreciate the test set! Two small things..
1) To make sure, you took the idea, but not the code, from stackoverflow, is that right? Just making sure there are no potential license issues here.
2) Erik's recent patches have allocated test set numbers 80-83, and that means there's a conflict now. If it's not too much work for you, could you rename yours to test 84? If you have no time available right now, I can also do it (although I prefer to take patches as is).
In any case, great work.

**Comment by boricj** [comment]
I used the second answer from the StackOverflow question, which has a 7-step list in plain English on how to deal with this specific issue. So there's no code to take from, and thus no license issue.

I do wonder what would happen if someone did something like ``posix_spawn_file_actions_addclose(&file_actions, 253);`` or otherwise interfere with file descriptors 252 and 253. The error handling could also be better, especially since there's no reason to have ``_exit(127);`` with the pipe solution. I guess I'll just wait for the code review to see if there are other issues before I start fixing them.

I'm rebasing right now, should push when it's done compiling and testing.

**Comment by dcvmoole** [comment]
It seems to me that it would be possible to avoid hardcoding the file descriptors. First, the 252 one need not be hardcoded at all - instead the child could just close(pfd[0]) right after the fork basically.

By the way, it appears that as is, upon success, 252 is never closed by the parent.

For the 253 one, the child could use the fcntl(F_DUPFD) system call to move around the file descriptor whenever it is about to be dup2'd over, basically with the following logic: 1) start by saving pfd[1] in some variable, let's call it "statusfd"; 2) right before dup2'ing onto a new file descriptor, check if the new file descriptor equals "statusfd"; if so, use `newstatusfd = fcntl(statusfd, F_DUPFD, statusfd+1);` to copy the open file to the next available file descriptor (do perform error checks here though! there may be no free file descriptors left), and `close(statusfd); statusfd = newstatusfd;` to complete the move. At that point it is safe to dup2 onto what used to be the old value of statusfd. At the end, you have "statusfd" to write an error to, and no 252/253 hardcoding is necessary at all.

As another aside, the parent's read() call to get the error should really check the return value. I believe POSIX doesn't promise that if the read gives you EOF, the contents of "error" are unchanged, while the code is in fact relying on that here.

**Comment by boricj** [comment]
I made all the suggested fixes.

I also gave up on trying to keep changes to newlib's ``posix_spawn.c`` at minimum, as most of the code ended up modified anyway. It looks slightly less out of place than before now.

**Comment by boricj** [comment]
I corrected the minors issues flagged in boricj/minix@f2bd9c8e84a7f7386ab557a2bc601f827fb9664b.

I also added some comments for clarity.

**Comment by dcvmoole** [comment]
I just made a few more small changes and submitted the whole thing to gerrit.

**Comment by dcvmoole** [comment]
Merged with the minor changes as 1facb0487c831b9ead49737c23d6f936b553009b.
